code/game/area/areas.dm

```diff
@@ -240,13 +323,62 @@
 		mouse_opacity = MOUSE_OPACITY_TRANSPARENT
 		updateicon()
 
+/**
+  * Raise a burglar alert for this area
+  *
+  * Close and locks all doors in the area and alerts silicon mobs of a break in
+  *
+  * Alarm auto resets after 600 ticks
+  */
+/area/proc/burglaralert(obj/trigger)
+	if(always_unpowered) //no burglar alarms in space/asteroid
+		return
+
+	//Trigger alarm effect
+	set_fire_alarm_effect()
+	//Lockdown airlocks
+	for(var/obj/machinery/door/DOOR in src)
+		close_and_lock_door(DOOR)
+
+	if(SSalarm.triggerAlarm("Burglar", src, cameras, trigger))
+		//Cancel silicon alert after 1 minute
+		addtimer(CALLBACK(SSalarm, /datum/controller/subsystem/alarm.proc/cancelAlarm, "Burglar", src, trigger), 600)
+
+/**
+  * Trigger the fire alarm visual affects in an area
+  *
+  * Updates the fire light on fire alarms in the area and sets all lights to emergency mode
+  */
+/area/proc/set_fire_alarm_effect()
+	fire = TRUE
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	for(var/alarm in firealarms)
+		var/obj/machinery/firealarm/F = alarm
+		F.update_fire_light(fire)
+	for(var/obj/machinery/light/L in src)
+		L.update()
+
+/**
+  * unset the fire alarm visual affects in an area
+  *
+  * Updates the fire light on fire alarms in the area and sets all lights to emergency mode
+  */
+/area/proc/unset_fire_alarm_effects()
+	fire = FALSE
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	for(var/alarm in firealarms)
+		var/obj/machinery/firealarm/F = alarm
+		F.update_fire_light(fire)
+	for(var/obj/machinery/light/L in src)
+		L.update()
+
 /area/proc/updateicon()
-	if((fire || eject || party) && (!requires_power||power_environ))//If it doesn't require power, can still activate this proc.
-		if(fire && !eject && !party)
+	if((eject || party) && (!requires_power||power_environ))//If it doesn't require power, can still activate this proc.
```

Comment:
The idea being, no more full area overlay for fire alarms, and instead it modifies the lighting of nearby lights and firedoors?

Reply:
Yep